sum_types_generator/lib/src/sum_type_spec.dart

```diff
@@ -118,7 +128,20 @@ CaseSpec _makeCaseSpec(
   if (ctor.parameters.length <= 1) {
     final caseType =
         ctor.parameters.isNotEmpty ? ctor.parameters.single.type : null;
-    return CaseSpec(name: ctor.name, type: makeCaseTypeSpec(caseType));
+
+    ParameterStyle style = ParameterStyle.positional;
+
+    if (ctor.parameters.isNotEmpty && ctor.parameters.single.isNamed) {
+      style = ParameterStyle.named;
+    }
```

Comment:
I'd recommend avoid mutable variables where (easily) possible. Here it is possible in a couple of ways.

One option is close to what's already written:

```dart
    final ParameterStyle style;

    if (ctor.parameters.isNotEmpty && ctor.parameters.single.isNamed) {
      style = ParameterStyle.named;
    } else {
      style = ParameterStyle.positional;
    }
```

The variable is `final`, but still can be initialized after the declaration.

Another option would be to use the ternary operator (essentially, an in-line `if`):

```dart
    final style = ctor.parameters.isNotEmpty && ctor.parameters.single.isNamed ? ParameterStyle.named : ParameterStyle.positional;
```

The first option is probably more readable.